File: src/main/java/me/crylonz/PermissionUtils.java
package me.crylonz;

import org.bukkit.entity.Player;

public class PermissionUtils {

    public static boolean hasOneOf(Player player, Permission[] permissions) {
        for(Permission permission: permissions) {
            if(player.hasPermission(permission.label))
                return true;
        }
        return false;
    }

    public static boolean hasAllOf(Player player, Permission[] permissions) {
        for(Permission permission: permissions) {
            if(!player.hasPermission(permission.label))
                return false;
        }
        return true;
    }

    public static boolean hasAdminOrAllOf(Player player, Permission[] permissions) {
        if(player.hasPermission(Permission.ADMIN.label))
            return true;
        return hasAllOf(player, permissions);
    }

    public static boolean hasAdminOrOneOf(Player player, Permission[] permissions) {
        if(player.hasPermission(Permission.ADMIN.label))
            return true;
        return hasOneOf(player, permissions);
    }

    public static boolean hasAdminOr(Player player, Permission permission) {
        if(player.hasPermission(Permission.ADMIN.label))
            return true;
        return player.hasPermission(permission.label);
    }

}

File: deadchest-core/src/main/kotlin/me/crylonz/deadchest/PermissionUtils.kt
package me.crylonz.deadchest

import me.crylonz.Permission
import me.crylonz.Permission.ADMIN
import org.bukkit.entity.Player

object PermissionUtils {

    var REMOVE_ALL = listOf(Permission.REMOVE_OWN, Permission.REMOVE_OTHER)
    val LIST_ALL = listOf(Permission.LIST_OWN, Permission.LIST_OTHER)

    fun hasOneOf(player: Player, permissions: List<Permission>): Boolean {
        return permissions.any { player.hasPermission(it.label) }
    }

    fun hasAllOf(player: Player, permissions: List<Permission>): Boolean {
        return permissions.all { player.hasPermission(it.label) }
    }

    fun hasAdminOrAllOf(player: Player, permissions: List<Permission>): Boolean {
        return if (player.hasPermission(ADMIN.label)) true else hasAllOf(player, permissions)
    }

    fun hasAdminOrOneOf(player: Player, permissions: List<Permission>): Boolean {
        return if (player.hasPermission(ADMIN.label)) true else hasOneOf(player, permissions)
    }

    fun hasAdminOr(player: Player, permission: Permission): Boolean {
        return if (player.hasPermission(ADMIN.label)) true else player.hasPermission(permission.label)
    }
}

